frame/bonded-finance/src/lib.rs

```diff
@@ -361,6 +361,7 @@ pub mod pallet {
 							&offer_account,
 							from,
 							VestingSchedule {
+								/// vesting_schedule_id: /// TODO: generate random id
```

Comment:
Could you check if there is a standard way to generate an ID for this in substrate?

Reply:
I just spoke with @vivekvpandya and we have 2 possible ways of handling this:
1. We replace the current storage map of `(AccountId, AssetId) -> Vec<VestingSchedule>` for something like `(AccountId, AssetId, count) -> VestingSchedule`, where `count` would be a `u128` that is incremented every time a `VestingSchedule` is added for a given pair of (`AccountId, AssetId`). We can then use `count` as a unique id for claiming rewards.

Reply:
I just spoke with @vivekvpandya and we have 2 possible ways of handling this:

1. We replace the current storage map of `(AccountId, AssetId) -> Vec<VestingSchedule>` for something like `(AccountId, AssetId, count) -> VestingSchedule`, where `count` would be a `u128` that is incremented every time a `VestingSchedule` is added for a given pair of (`AccountId, AssetId`). We can then use `count` as a unique id for claiming rewards. This would require an extra storage of `(AccountId, AssetId) -> count`.

2. Similarly, we add a `count` id to the `VestingSchedule` structure, and we keep the `VestingSchedules` storage as it is, while also adding the extra storage of `(AccountId, AssetId) -> count`. This would keep things simpler IMO because we wouldn't need to refactor too much.

Reply:
Tracking a count is ok, we can track it separately and use that as the ID inside the VestingSchedule struct.

> while also adding the extra storage of (AccountId, AssetId) -> count

I think this doesn't need to be a map no? We can just have a counter in storage.

Reply:
Yes, right 👌🏼